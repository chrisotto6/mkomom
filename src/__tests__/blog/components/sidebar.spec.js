import React from 'react'
import { create } from 'react-test-renderer'
import { StaticQuery } from 'gatsby'
import Sidebar from '../../../blog/components/sidebar'

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      site: {
        siteMetadata: {
          siteUrl: `Default Starter`,
        },
      },
    })
  )
})

describe('Sidebar', () => {
  it('renders correctly', () => {
    const tree = create(<Sidebar desc="test" img="test.jpg" location="test location" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
