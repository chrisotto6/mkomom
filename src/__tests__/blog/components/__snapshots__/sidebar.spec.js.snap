// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders correctly 1`] = `
<div
  className="dn db-l"
  style={
    Object {
      "gridArea": "sidebar",
    }
  }
>
  <div
    className="w3 bg-dark-gray flex flex-wrap"
    style={
      Object {
        "position": "sticky",
        "top": "4rem",
      }
    }
  >
    <a
      className="w-100 h3 flex items-center justify-center b near-white"
      href="https://pinterest.com/pin/create/button/?url=Default%20Starterundefined&description=test&media=Default%20Startertest.jpg"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 384 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M204 6.5C101.4 6.5 0 74.9 0 185.6 0 256 39.6 296 63.6 296c9.9 0 15.6-27.6 15.6-35.4 0-9.3-23.7-29.1-23.7-67.8 0-80.4 61.2-137.4 140.4-137.4 68.1 0 118.5 38.7 118.5 109.8 0 53.1-21.3 152.7-90.3 152.7-24.9 0-46.2-18-46.2-43.8 0-37.8 26.4-74.4 26.4-113.4 0-66.2-93.9-54.2-93.9 25.8 0 16.8 2.1 35.4 9.6 50.7-13.8 59.4-42 147.9-42 209.1 0 18.9 2.7 37.5 4.5 56.4 3.4 3.8 1.7 3.4 6.9 1.5 50.4-69 48.6-82.5 71.4-172.8 12.3 23.4 44.1 36 69.3 36 106.2 0 153.9-103.5 153.9-196.8C384 71.3 298.2 6.5 204 6.5z"
        />
      </svg>
    </a>
    <a
      className="w-100 h3 flex items-center justify-center b near-white"
      href="https://www.facebook.com/sharer/sharer.php?u=Default%20Starterundefined"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 320 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
        />
      </svg>
    </a>
    <a
      className="w-100 h3 flex items-center justify-center b near-white"
      href="mailto:?&body=Default%20Starterundefined"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
        />
        <polyline
          points="22,6 12,13 2,6"
        />
      </svg>
    </a>
  </div>
</div>
`;
